apps/wallet/src/ui/app/pages/qredo-connect/hooks.ts

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { useQuery } from '@tanstack/react-query';
+
+import { useBackgroundClient } from '../../hooks/useBackgroundClient';
+import { useQredoAPI } from '../../hooks/useQredoAPI';
+import { type GetWalletsParams } from '_src/shared/qredo-api';
+
+export function useQredoUIPendingRequest(requestID?: string) {
+    const backgroundClient = useBackgroundClient();
+    return useQuery({
+        queryKey: ['qredo-connect', 'pending-request', requestID],
+        queryFn: async () =>
+            await backgroundClient.fetchPendingQredoConnectRequest(requestID!),
+        staleTime: 0,
+        refetchInterval: 1000,
+        enabled: !!requestID,
+        meta: { skipPersistedCache: true },
+    });
+}
+
+export function useFetchQredoAccounts(
+    qredoID?: string,
```

Comment:
does `qredoId` need to be optional / do you see a case where it might be optional? I was just referencing https://github.com/MystenLabs/sui/pull/11223/files and it looks like we only use this hook once but qredoId is always defined

Reply:
Yeah you are right I will update it. Usually though ids like this is useful to be optional or maybe null as well